### ISSUE TITLE
[Bug] Unexpected result when writing invalid line protocol 

### ISSUE THREAD

**Issue by zipper-meng** [cnosdb/cnosdb#1817]
### Describe the bug

I'm trying to write invalid line protocol, the request is failed but server responses nothing.

### Trying to write line protocol data with invalid tag: 

Write invalid line protocol:

```sh
curl -X POST -u 'a:b' 'http://127.0.0.1:8902/api/v1/write?db=public' -d "log,host=domain.1,service=service-1,message=\"{\\\"time\\\":1702277964989014000,\\\"level\\\":\\\"info\\\",\\\"message\\\":\\\"Creating tenant\\\"}\" message=\"{\\\"time\\\":null,\\\"level\\\":\\\"info\\\",\\\"message\\\":\\\"Creating tenant\\\"}\" 1702277964989014000"
curl: (52) Empty reply from server
```

Server panics:

```log
The application panicked (crashed).
Message:  begin <= end (71 <= 39) when slicing `host=domain.1,service=service-1,message="{\"time\":1702277964989014000,\"level\":\"info\",\"message\":\"Creating tenant\"}" message="{\"time\":null,\"level\":\"info\",\"message\":\"Creating tenant\"}" 1702277964989014000`
Location: common/protocol_parser/src/lib.rs:237
```

### There is also simplified data to reproduce:

```sh
curl -X POST -u 'root:' 'http://127.0.0.1:8902/api/v1/write?db=public' -d "log,host=domain.1,service=service-1,a,b,c d=1"
curl: (52) Empty reply from server
```

The panic log:

```log
The application panicked (crashed).
Message:  begin <= end (32 <= 21) when slicing `host=domain.1,service=service-1,a,b,c d=1`
Location: common/protocol_parser/src/lib.rs:237
```

### Expected behavior

No panic, give me my mistake.

**Comment by zipper-meng** [comment]
We should also fix this on LTS/2.3.

**Comment by Benxiaohai001** [comment]
Done
